### PR TITLE
Fix System.Globalization.Native build on Big Sur

### PR DESCRIPTION
**Pull request by safern**
Port of: https://github.com/dotnet/runtime/pull/39900 to be able to build globalization shim on big sur.

cc: @danmoseley @tarekgh

### CODE REVIEW

**Review by tarekgh — APPROVED** [review]

**Comment by tarekgh** [comment]
@safern looks this change causing some compile errors:

```
[  1%] Linking C shared library System.Globalization.Native.dylib
ld: unknown option: --build-id=sha1
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/corefx/System.Globalization.Native/System.Globalization.Native.dylib] Error 1
make[1]: *** [src/corefx/System.Globalization.Native/CMakeFiles/System.Globalization.Native.dir/all] Error 2
```

**Comment by safern** [comment]
Yes now I'm looking at that error, thanks 😊

**Comment by jeffschwMSFT** [comment]
@safern the ci finished with errors.  
There appear to be a few issues:
MacOS build failed: https://dev.azure.com/dnceng/public/_build/results?buildId=1188408&view=logs&j=5d252662-176a-5e67-1101-74d1a75ad6cb&t=220080c0-1a38-5406-f449-63dd4d7c3755&l=389
System.Tex.RegularExpressions test was killed: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-coreclr-refs-pull-28181-merge-53f25af7c12b43648d/System.Text.RegularExpressions.Tests/console.14252b93.log?sv=2019-07-07&se=2021-07-05T18%3A50%3A39Z&sr=c&sp=rl&sig=C8mTAXbM6SrOhCFJhw%2FTk6ppx3qCUtvaPTi%2FsPTfaow%3D
System.Net failures due to cert: https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-coreclr-refs-pull-28181-merge-53f25af7c12b43648d/System.Net.Primitives.UnitTests.Tests/console.d00fe4f0.log?sv=2019-07-07&se=2021-07-05T18%3A50%3A39Z&sr=c&sp=rl&sig=C8mTAXbM6SrOhCFJhw%2FTk6ppx3qCUtvaPTi%2FsPTfaow%3D


cc @wtgodbe

**Comment by safern** [comment]
The build failure I'm hitting on OSX was introduced on: https://github.com/dotnet/coreclr/pull/28143 `--build-id` flag should only be set on Linux and FreeBSD. cc: @ivdiazsa

**Comment by jeffschwMSFT** [comment]
@mangod9

**Comment by safern** [comment]
It seems like this line should not be added: https://github.com/dotnet/coreclr/blob/3d4e15b1f117ae42f7c93f61ff1fed73e6b229ff/configurecompiler.cmake#L344

Those flags are already set for FreeBSD and Linux down on that file:
https://github.com/dotnet/coreclr/blob/3d4e15b1f117ae42f7c93f61ff1fed73e6b229ff/configurecompiler.cmake#L362

The original commit that the PR in question was porting didn't include that line:
https://github.com/dotnet/coreclr/pull/26323/files#diff-bcae6f2abcd5726994b49517ade5a2f939f873deac34644642e8e9f710cef757R344

So it seems like that line was added accidentally.

**Comment by safern** [comment]
I pushed a commit to make `configurecompiler.cmake` match the change on release/5.0 that was ported.

File: configurecompiler.cmake [hunk 1]
```diff
@@ -341,8 +341,7 @@ elseif (CLR_CMAKE_PLATFORM_UNIX)
   # This linker option causes executables we build to be marked as containing
   # position independent code.
   # It is necessary to make ASLR work for executables.
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--build-id=sha1 -Wl,-z,relro,-z,now")
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pie -Wl,-z,relro,-z,now")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -pie")
```
Comment (janvorli) [inline configurecompiler.cmake:344]: This removes the `-Wl,-z,relro,-z,now` but it is needed for Linux. Can you please add it to the  CMAKE_SHARED_LINKER_FLAGS and CMAKE_EXE_LINKER_FLAGS in the Linux if block below?

Comment (safern) [reply configurecompiler.cmake:344]: Yes. Do we need it on FreeBSD as well?

Comment (janvorli) [reply configurecompiler.cmake:344]: No, just Linux (looking at the flags set in main)

Comment (safern) [reply configurecompiler.cmake:344]: Thanks.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Review by mangod9 — APPROVED** [review]

**Review by jeffschwMSFT — APPROVED** [review]
Approved.

**Comment by jeffschwMSFT** [comment]
Merging to unblock the build

**Pull request merged**